workspaces/ui-v2/src/contexts/analytics/AnalyticsStore.tsx

```diff
@@ -17,8 +25,9 @@ type AnalyticsMetadata = {
   clientId: string;
   clientAgent: string;
   apiName: string;
+  specId?: string;
```

Comment:
I made it optional because the `getMetadata` functions are expected to return this `AnalyticsMetadata` type, despite not being expected to actually return the `specId` (since it's automatically gotten and injected into the metadata in `AnalyticsStore.tsx`). I [changed the types around a bit](https://github.com/opticdev/optic/pull/1083/commits/7e7b62ebfb0d08e6740cbc72a2229a8db449329a), let me know if you'd prefer it a different way

Reply:
Simplicity might be better to just have a default specId of `anonSpecId` - it's similar to having undefined, but also gives us a way to trace / know our analytics aren't working well / there's something not being set

Reply:
Simplicity as something different from the commit I linked? Or are you just saying I should add a default spec ID to make super clear when it's not available?